File: NQueen.java
package nqueen.java;

public class NQueen{

   private int[][] position;
   private int n;
   private int curQueen;

   public NQueen(int n){
      this.n = n;
      position=new int[n][2];
      putQueens(0, 0, 0);
   }
   private void putQueens(int x, int y, int queen){
      position[queen][0]=x;
      position[queen][1]=y;
      queen++;
      curQueen = queen;
      check(0, y+1, queen);
   }
   private void check(int x, int y, int queen){
      int[] result = isSafe(x, y, true);
      if (queen==n){
         return;
      } if (result[0]>-1){
         x = result[0];
         y = result[1];
         putQueens(x, y, queen);
      } else {
         int prevQueen = queen - 1;
         curQueen = prevQueen;
         int prevX = position[prevQueen][0];
         int prevY = position[prevQueen][1];
         check(prevX+1, prevY, prevQueen);
      }
   }
   private int[] isSafe(int x, int y, boolean safe){
      if (x >= n){
         int[] arr = {-1, -1};
         return arr;
      } else {
         for (int i=0; i < curQueen; i++){
            int posX = position[i][0];
            int posY = position[i][1];
            if (x==posX || Math.abs(y-posY)==Math.abs(x-posX)){
               return isSafe(x + 1, y, true);
            }
         }
      }
      int[] arr = {x, y};
      return arr;
   }
   public void next(){
      int x = position[n-1][0]+1;
      int y = position[n-1][1];
      curQueen--;
      check(x, y, n-1);
   }
   public String getN(){
      return String.format("%d", n);
   }
   public String getPosition(){
      String result=n+" Queens:\nx   y\n";
      for (int i=0; i < position.length; i++){
         result+=Integer.toString(position[i][0])+" | "+Integer.toString(position[i][1])+"\n";
      }
      return result;
   }
}

File: nQueen.py

n =13
position = []

def isSafe(x, y, position, safe=True):
    if x >= n :
        return []
    else:
        for i in range(len(position)):
            if (x == position[i][0] or abs(y - position[i][1]) == abs(x - position[i][0])):
            #abs(y - position[i][1]) == abs(x - position[i][0]) same as y - x == position[i][1] - position[i][0] or y + x == position[i][1] + position[i][0]):
                safe = False
                break
            else:
                safe = True
    if not safe:
        return isSafe(x + 1, y, position, safe)
    return [x, y]

def logs(queen, position, x, y):
    print('queen', queen+1)
    print('pos', position)
    print('_________________________')

def check(x, y, queen, position, result=[]):
    result = isSafe(x, y, position)
    if queen == n:
        return True
    if len(result) > 1:
        x = result[0]
        y = result[1]
        putQueens(x, y, queen)
    else:
        prevQueen = queen - 1
        prevX = position[prevQueen][0]
        prevY = position[prevQueen][1]
        position.pop()
        check(prevX+1, prevY, prevQueen, position)
def putQueens(x=0, y=0, queen=0):
    position.append([x, y])
    logs(queen, position, x, y)
    check(0, y+1, queen+1, position)

putQueens()
